File: motion-canvas-scenes/src/theme.ts
/**
 * Theme loader for Motion Canvas scenes.
 * Reads theme.json from the project root for cross-engine visual consistency.
 */

import themeData from '../../theme.json';

export const theme = {
  colors: themeData.colors,
  fonts: themeData.fonts,
  spacing: themeData.spacing,
  animation: themeData.animation,
  canvas: themeData.canvas,
} as const;

// Convenience accessors
export const colors = theme.colors;
export const fonts = theme.fonts;
export const spacing = theme.spacing;
export const animation = theme.animation;

File: manim-scenes/theme.py
"""Theme loader for Manim scenes — reads theme.json for cross-engine consistency."""

import json
import os

_theme_path = os.path.join(os.path.dirname(__file__), '..', 'theme.json')

with open(_theme_path) as f:
    _theme = json.load(f)

colors = _theme['colors']
fonts = _theme['fonts']
spacing = _theme['spacing']
animation = _theme['animation']
canvas = _theme['canvas']
